### ISSUE TITLE
Missing files

### ISSUE THREAD

**Issue by luclemot** [aphp/eds-pseudo#5]
## Description

It appears that the repository is missing two files : _infer.py_ and _train.py_. These files are mentioned in the _project.yml_ file.
Have they been renamed, or is there another way to train the eds-pseudo module than what is mentioned in the docs ?

## How to reproduce the bug

After following the quickstart tutorial from the [github.io](https://aphp.github.io/eds-pseudo/0.2.0/quickstart/#installation), I can't seem to be able to train the model on my dataset.

The error I get when running `dvc repro` is :

`ERROR: failed to reproduce 'xp': [Errno 2] No such file or directory: 'eds-pseudo/scripts/infer.py'`

I suspect the same error would occur with the _train.py_ file.

## Your Environment

- Operating System: Ubuntu
- Python Version Used: 3.10
- EDS-Pseudo Version Used: 0.2.0

**Comment by percevalw** [comment]
Hi @luclemot, thanks for the feedback ! I indeed forgot to commit the `infer.py` script. As for the `train.py` script, it is not needed since spaCy handles the training loop and optimization process. I've put some fixes here https://github.com/aphp/eds-pseudo/pull/6, do you mind trying again on this branch ?

**Comment by luclemot** [comment]
Hi,
Thank you for your reactivity !
I'm now facing another issue : I can't seem to find the right data structure for my annotated dataset (which fields should I include, and in what format?).
Moreover, it seems like the data split isn't included in the spacy workflow. Is there a best practice for this dataset split ?
Thank you for your help !

**Comment by percevalw** [comment]
Indeed, this is also missing from the documentation ! I'll add a page about this soon. How is your original data formatted ? 

An annotated document should look like this:

```python
doc = {
    "note_id": "my-note_id",
    "note_text": "Jean-Paul habite à Cergy",
    "entities": [
        {"start": 0, "end": 9, "label": "PRENOM"},
        {"start": 19, "end": 24, "label": "VILLE"},
    ],
    "context": {
        "ZIP": ["75001"],
        "SECU": ["123456789011"],
        "IPP": ["8000000000"],
        "NDA": ["123454321"],
        "TEL": ["060102030405"],
        "VILLE": ["CERGY"],
        "ADRESSE": ["1 RUE DE LA VILLE"],
        "NOM": ["DUCHAMP"],
        "PRENOM": ["Jean Paul"],
        "NOM_NAISS": ["DUCHAMP"],
        "EMAIL": [],
    },
}
```
the `context` field can be empty, if no info is known about the subject of the document.

Given a list of objects like this in Python, you can write them to a JSONL file using the following snippet:
```python
import json

docs = [doc, ...]

with open("data/dataset/train.jsonl", 'w') as f:
    for item in docs:
        f.write(json.dumps(item) + "\n")
```

This file will then be converted to a spaCy binary file in the `script/convert.py`.

About the data split, it depend on the quantity of documents. My rule of thumb is that 10-20% of documents (at least 40-50 documents) should be used for validation, but this can vary a lot depending on the project, type of data, etc.

Also, any change you've might have made to the documentation is welcome :) do not hesitate to submit a PR.

**Comment by percevalw** [comment]
I've merged #6 so this should fix the aforementioned issues, don't hesitate to open another issue if something comes up !
